docs/deployment.md

```diff
@@ -0,0 +1,95 @@
+# Deployment
+
+## Deploying locally
+
+Your function can run locally either by starting it through Visual
+Studio or by running `dotnet run` from the command line. If your
+project contains multiple functions, you must specify which one you
+want to host either via command line arguments or environment
+variables. The port to serve on can also be specified via the
+command line or environment variables.
+
+See the [README](../README.md) for configuration details.
+
+## Deploying to Cloud Run
+
+Running the Functions Framework is equivalent to running a regular
+ASP.NET Core application as far as infrastructure is concerned. This
+means you can run functions on Cloud Run with ease.
+
+There are detailed instructions for C# available as part of the
+the Cloud Run documentation:
+
+- [Quick start for C#](https://cloud.google.com/run/docs/quickstarts/build-and-deploy#c)
+- [Codelab for C#](https://codelabs.developers.google.com/codelabs/cloud-run-hello-csharp/index.html)
+
+The instructions apply to functions in exactly the same way as
+normal, with the same Dockerfile configuration. The `PORT`
+environment variable will be supplied by Cloud Run. If your project
+contains multiple functions, the simplest approach is to specify an
+environment variable in your Dockerfile, with a line like this:
+
+```text
+ENV FUNCTION_TARGET HelloFunctions.Function
+```
+
+## Deploying to Google Cloud Functions
+
+> **NOTE:**  
+> The `dotnet3` runtime is currently available on an "allow
+> list" basis. Deployment will fail unless your project has been
+> specifically included in the list.
+
+Deployment to Google Cloud Functions is easy with the `gcloud`
+command line. See the [general deployment
+documentation](https://cloud.google.com/functions/docs/deploying)
+for more detailed information. In general, deploying a .NET
+Functions Framework function follows the same procedure as deploying
+a function written in every language. The .NET-specific aspects are:
+
+- The runtime should be specified as `dotnet3`
+- The entry point should be specified as the name of the function
+  type, including namespace (e.g. HelloFunctions.Function)
+
+The command line options are used to specify how the function is
+triggered.
+
+### HTTP functions
```

Comment:
Instead of duplicating the information on https://cloud.google.com/functions/docs/deploying/filesystem, could you just link to that page? With the details about the runtime name and the entrypoint conditions, of course, as that documentation is language-agnostic.

Reply:
The aim was to give a smaller amount of information that readers could use to get started without the extra detail on that page. It's a little less daunting that way, IMO. Definitely happy to tweak though (e.g. make this even barer, and/or update the link in the previous paragraph to the link you suggested, instead of the more general one). Also happy to be overridden entirely.

Reply:
I just think this will get out of date fairly quickly -- it won't be on anyone's radar to update, and so it's easier if it's all in one place. 

Barer would be fine, I think, with the reference to the more in-depth documentation.

Reply:
I've removed a little bit, and updated the link - but most of what's left felt like it's "stuff that might not be entirely obvious if looking at the detailed docs". (It took me a while to understand the exact syntax for the Firestore trigger, for example.)

Another option that might appeal more would be to create a sample function for each of the legacy event types, and put a README.md file with a sample command line to deploy it into each of the sample directories.